pkg/genall/genall.go

```diff
@@ -103,7 +103,7 @@ func (g GenerationContext) WriteYAML(itemPath string, objs ...interface{}) error
 		if err != nil {
 			return err
 		}
-		n, err := out.Write(append([]byte("\n---\n"), yamlContent...))
+		n, err := out.Write(append([]byte("---\n"), yamlContent...))
```

Comment:
is `yaml.Marshal` guaranteed to produce a newline at the end?

Reply:
(i.e. is this safe to not have the extra newline?)

Reply:
I think we actually have a helper burried in the k8s libraries for this

Reply:
(nope, just a decoder)

Reply:
I think `yaml.Marshal` is guaranteed to produce a newline